factory/usr/lib/systemd/system/sysroot.mount

```diff
@@ -2,10 +2,16 @@
 DefaultDependencies=no
 Before=initrd-root-fs.target
 
+After=detect-uc-sysroot.service
+Wants=detect-uc-sysroot.service
+After=detect-classic-sysroot.service
+Wants=detect-classic-sysroot.service
 After=run-mnt-base.mount
 After=snap-initramfs-mounts.service
 
+AssertPathIsSymbolicLink=/run/mnt/sysroot
+
 [Mount]
-What=/run/mnt/base
+What=/run/mnt/sysroot
```

Comment:
Maybe it would be good to add a comment, something like:
```
# /run/mnt/sysroot is a link to either the base on UC or to the rootfs on classic,
# depending on whether snap-bootstrap mounts or not /run/mnt/base, it will be
# one or the other.
```

Reply:
Done